upload/upload_test.go

```diff
@@ -135,21 +135,6 @@ func (suite *TestSuite) TestcreateFilePaths() {
 	assert.ErrorContains(suite.T(), err, msg)
 }
 
-func (suite *TestSuite) TestFormatUploadFilePath() {
-
-	unixPath := "a/b/c.c4gh"
-	testPath := filepath.Join("a", "b", "c.c4gh")
-	uploadPath := formatUploadFilePath(testPath)
-
-	assert.Equal(suite.T(), unixPath, uploadPath)
-
-	weirdPath := "a/b:/c;.c4gh"
-	goodPath := "a/b_/c_.c4gh"
-	formattedPath := formatUploadFilePath(weirdPath)
-
-	assert.Equal(suite.T(), goodPath, formattedPath)
-}
```

Comment:
This test should be replaced by one where ensure files with invalid characters result in an error

Reply:
I moved out the check for invalid chars to [helpers](https://github.com/NBISweden/sda-cli/pull/287/files#diff-ba41f3a7938a59c0335de7decc4304f93218d516ce4b6340665a6e7b001e6230R402) so that it can be tested on its own. Running [`upload`](https://github.com/NBISweden/sda-cli/pull/287/files#diff-3ba32a4f84fefb210b231b3311ecc0af4f2805b0f7278ee53c1f4044ad2f603cR490) is only tested for linux, since windows won't allow having these characters in a filepath, and the test will hence fail on the fail creation.